go-controller/pkg/ovn/namespace.go

```diff
@@ -330,26 +330,38 @@ func (oc *DefaultNetworkController) getAllHostNamespaceAddresses() []net.IP {
 				// skip hybrid overlay nodes
 				continue
 			}
-			hostSubnets, err := util.ParseNodeHostSubnetAnnotation(node, types.DefaultNetworkName)
+			hostNetworkIPs, err := oc.getHostNamespaceAddressesForNode(node)
 			if err != nil {
-				klog.Warningf("Error parsing host subnet annotation for node %s (%v)",
-					node.Name, err)
-			}
-			for _, hostSubnet := range hostSubnets {
-				mgmtIfAddr := util.GetNodeManagementIfAddr(hostSubnet)
-				ips = append(ips, mgmtIfAddr.IP)
-			}
-			// for shared gateway mode we will use LRP IPs to SNAT host network traffic
-			// so add these to the address set.
-			lrpIPs, err := util.ParseNodeGatewayRouterLRPAddrs(node)
-			if err != nil {
-				klog.Errorf("Failed to get join switch port IP address for node %s: %v", node.Name, err)
-			}
-
-			for _, lrpIP := range lrpIPs {
-				ips = append(ips, lrpIP.IP)
+				klog.V(5).Infof("Error parsing annotation for node %s: %v", node.Name, err)
 			}
+			ips = append(ips, hostNetworkIPs...)
 		}
 	}
 	return ips
 }
+
+func (oc *DefaultNetworkController) getHostNamespaceAddressesForNode(node *kapi.Node) ([]net.IP, error) {
+	var ips []net.IP
+	hostSubnets, err := util.ParseNodeHostSubnetAnnotation(node, types.DefaultNetworkName)
+	if err != nil {
+		klog.Warningf("Error parsing host subnet annotation for node %s: %v",
```

Comment:
so the past behaviour was to warn and not return error and if you fix the above comment of mine, this means from here when you return the error it will percolate up where you will print it. Do you really want to print and return the errors here in this function?

Reply:
Changed it to only return the error instead of both printing and returning.